Description: Revert incorrect off-by-one fix when embedding in MBR
 A 62-sector core image should fit before end == 63.
Author: Colin Watson <cjwatson@ubuntu.com>
Origin: upstream, http://bazaar.launchpad.net/~vcs-imports/grub/grub2-bzr/revision/4586
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1051154
Forwarded: not-needed
Applied-Upstream: http://bazaar.launchpad.net/~vcs-imports/grub/grub2-bzr/revision/4586
Last-Update: 2012-09-19

Index: b/grub-core/partmap/msdos.c
===================================================================
--- a/grub-core/partmap/msdos.c
+++ b/grub-core/partmap/msdos.c
@@ -316,14 +316,14 @@
 	break;
     }
 
-  if (end >= *nsectors + 2)
+  if (end >= *nsectors + 1)
     {
       unsigned i, j;
       char *embed_signature_check;
       unsigned int orig_nsectors, avail_nsectors;
 
       orig_nsectors = *nsectors;
-      *nsectors = end - 2;
+      *nsectors = end - 1;
       avail_nsectors = *nsectors;
       if (*nsectors > max_nsectors)
 	*nsectors = max_nsectors;
